exonum-java-binding/core/rust/src/testkit/mod.rs

```diff
@@ -213,3 +214,122 @@ fn create_java_keypair<'a>(
         &[secret_key_byte_array.into(), public_key_byte_array.into()],
     )
 }
+
+// Converts Java array of `TestKitServiceInstances` to vector of `InstanceConfig`.
+//
+// `TestKitServiceInstances` representation:
+//      String artifactId;
+//      String artifactFilename;
+//      ServiceSpec[] serviceSpecs;
+fn instance_configs_from_java_array(
+    env: &JNIEnv,
+    service_artifact_specs: jobjectArray,
+) -> JniResult<Vec<InstanceConfig>> {
+    let mut instance_configs = vec![];
+    let num_artifacts = env.get_array_length(service_artifact_specs)?;
+    for i in 0..num_artifacts {
```

Comment:
As @dmitry-timofeev mentioned in https://github.com/exonum/exonum-java-binding/pull/1161/files#r333421566, this loop creates some amount of local references which can cause problems with large `num_artifacts`.

Reply:
Indeed, missed that...